### PR TITLE
User guide improvements

### LINKED ISSUE #4653: Description of some features missing the the documentation
Section 2.3 of the documentation lists the features that can be activated at compile time.

* ROTATIONAL_INERTIA - not described, unclear how it differs from rotation
* ENGINE - not described, the name is too generic and could mean practically anything, possibly blocking name space for other features in the future
* PARTICLE_ANISOTROPY  - not described
* VIRTUAL_SITES and VIRTUAL_SITES_INERTIALESS_TRACERS - neither listed nor described, only VIRTUAL_SITES_RELATIVE is described
* EXCLUSIONS - presumably inaccurate description because it suggests that all short-ranged interactions within a molecule are excluded
* DIPOLAR_DIRECT_SUM - described in the docs but not available in myconfig-sample.hpp

Additionally, it might be helpful to keep the same order and groupping of features in the docs and in myconfig-sample.hpp

Listing dependencies between various features might be helpful too (e.g. VIRTUAL_SITES cannot be activated without MASS)

### PR DESCRIPTION
**Pull request by jngrad**
Fixes #4653

Description of changes:
- fix errors in installation instructions: `make -j` without a number will start more processes than threads
- document particle exclusions and compile-time features 
- improve bibliography

### CODE REVIEW

File: doc/bibliography.bib [hunk 1]
```diff
@@ -156,27 +158,27 @@ @Article{bindgen21a
 }
 
 @Article{brady88a,
-  author  = {J. F. Brady and G. Bossis},
-  title   = {Stokesian Dynamics},
+  author  = {Brady, John F and Bossis, Georges},
```
Comment (pkreissl) [inline doc/bibliography.bib]: Missing dot after F

File: doc/bibliography.bib [hunk 2]
```diff
@@ -702,14 +710,14 @@ @book{moshier89a
...
-volume={68},
-year={1996},
-doi={10.1351/pac199668122193},
+  author    = {Moss, G. P.},
+  title     = {Basic terminology of stereochemistry},
+  journal   = {Pure and Applied Chemistry},
+  year      = {1996},
+  volume    = {68},
+  number    = {12},
+  pages     = {2193-2222},
```
Comment (pkreissl) [inline doc/bibliography.bib]: 2193--2222

File: doc/bibliography.bib [hunk 3]
```diff
@@ -832,31 +839,31 @@ @Article{rohm12a
...
   author = {Michael Rubinstein and Ralph H. Colby},
   address = {Oxford, UK},
 }
 
-@ARTICLE{schatzel88a,
-  author = {Sch\"{a}tzel, K. and Drewel, M. and Stimac, S},
-  title = {Photon Correlation Measurements at Large Lag Times: Improving Statistical Accuracy},
+@Article{schatzel88a,
+  author = {Sch\"{a}tzel, Klaus and Drewel, Martin and Stimac, Sven},
+  title = {Photon correlation Measurements at Large Lag Times: Improving Statistical Accuracy},
```
Comment (pkreissl) [inline doc/bibliography.bib]: Why small `c'? Screws up title case (which also is used for formatting the title on on the referenced webpage via doi)

File: doc/bibliography.bib [hunk 4]
```diff
@@ -984,45 +994,47 @@ @Article{tironi95a
...
+  pages = {154723},
   doi = {10.1063/1.2790428},
-  pages = {154723}
 }
 
-@ARTICLE{tyagi08a,
+@Article{tyagi08a,
   author = {Sandeep Tyagi and Axel Arnold and Christian Holm},
-  title = {Electrostatic layer correction with image charges: A linear scaling method to treat slab 2{D} + h systems with dielectric interfaces},
+  title = {Electrostatic layer correction with image charges: {A} linear scaling method to treat slab 2{D} + h systems with dielectric interfaces},
```
Comment (pkreissl) [inline doc/bibliography.bib:1022]: do you want to make the `h' into a math environment (or italic), as in the original title?

File: doc/sphinx/electrostatics.rst [hunk 5]
```diff
@@ -269,9 +269,9 @@ Electrostatic Layer Correction (ELC)
 systems. It can account for different dielectric jumps on both sides of the
 non-periodic direction. In more detail, it is a special procedure that
 converts a 3D electrostatic method to a 2D method in computational order N.
-The periodicity has to be set to (1 1 1). *ELC* cancels the electrostatic
+The periodicity has to be set to ``(True, True, True)``. *ELC* cancels the electrostatic
 contribution of the periodic replica in **z-direction**. Make sure that you
```
Comment (pkreissl) [inline doc/sphinx/electrostatics.rst]: z in z-direction should probably also be math environment/italic

File: doc/sphinx/installation.rst [hunk 6]
```diff
@@ -449,15 +455,17 @@ General features
...
 
    .. seealso:: :meth:`espressomd.particle_data.ParticleHandle.add_exclusion`
 
 -  ``BOND_CONSTRAINT`` Turns on the RATTLE integrator which allows for fixed lengths bonds
    between particles.
 
--  ``VIRTUAL_SITES`` Allows to set particles wthout meaningful mass, whose position is generally fixed in the simula
-   box or fixed to other particles.
+-  ``VIRTUAL_SITES`` Allows the creation of pseudo-particles whose forces,
+   torques and orientations can be transferred to real particle.
```
Comment (pkreissl) [inline doc/sphinx/installation.rst]: I'd add an Oxford comma...

File: doc/sphinx/inter_non-bonded.rst [hunk 7]
```diff
@@ -48,6 +48,18 @@ For many non-bonded interactions, it is possible to artificially cap the
 forces, which often allows to equilibrate the system much faster. See
 the subsection :ref:`Capping the force during warmup` for more details.
 
+It is possible to exclude particle pairs from the non-bonded interaction
+calculation. For example::
+
+    system.part.by_id(0).exclusions = [1, 2]
+
+exclude short-range interactions of the particle pairs ``0 <-> 1`` and ``0 <-> 2``.
```
Comment (pkreissl) [inline doc/sphinx/inter_non-bonded.rst]: to exclude/exclude**s**

**Review by pkreissl — COMMENTED** [review]
Just some typo stuff...

File: doc/bibliography.bib (hunk 4 above)
Comment (jngrad) [reply doc/bibliography.bib:1022]: Unfortunately, the BibTeX processor used by Sphinx currently doesn't handle math environments.

**Review by reinaual — APPROVED** [review]

**Pull request merged**